sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -98,8 +100,22 @@ case class DataSourceResolution(
 
     case DropViewStatement(AsTableIdentifier(tableName), ifExists) =>
       DropTableCommand(tableName, ifExists, isView = true, purge = false)
+
+    case u @ UnresolvedRelation(CatalogObjectIdentifier(Some(catalog), ident)) =>
```

Comment:
Why in this rule, instead of `FindDataSourceTable`?

Reply:
All DSv2 resolutions are kept in DataSourceResolution that requires LookupCatalog mixin.

Reply:
```
      new FindDataSourceTable(session) +:
        new ResolveSQLOnFile(session) +:
        new FallBackFileSourceV2(session) +:
        DataSourceResolution(conf, session.catalog(_)) +:
```

I do not know who make this decision, but semantically, this could be different. one is before the rule ResolveSQLOnFile; another is after the rule ResolveSQLOnFile

Reply:
Thanks for the suggestion. Let me give it some thought.

Reply:
Now that #24416 is merged, this rule can be in the analyzer. It was only here because DataSourceV2Relation was not available in Catalyst and, as John mentioned, this is where we are placing rules that convert to logical plan nodes that are specific to DataSource.

Reply:
When attempting to move `DataSourceV2Relation` to Catalyst, I discovered [SPARK-27961](https://issues.apache.org/jira/browse/SPARK-27961)
"DataSourceV2Relation should not have refresh method".

Reply:
@gatorsmile, looks like `DataSourceV2Relation` was not moved into catalyst in #24416 as expected. I think it makes sense to leave this rule here so that fixing #24401 and moving the relation to Catalyst doesn't block this PR.